integration/ctr/oci_hooks_tests.bats

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env bats
+
+IMAGE="${IMAGE:-docker.io/library/busybox:latest}"
+CONTAINER_NAME="${CONTAINER_NAME:-kataocihookstest}"
+CTR_RUNTIME="${CTR_RUNTIME:-io.containerd.kata.v2}"
+
+set -o errexit
+set -o nounset
+set -o pipefail
+set -o errtrace
+
+readonly KATA_CONFIG="${KATA_CONFIG:-/etc/kata-containers/configuration.toml}"
+readonly KATA_CONFIG_BACKUP="$KATA_CONFIG.backup"
+readonly DEFAULT_KATA_CONFIG="/usr/share/defaults/kata-containers/configuration.toml"
+
+function teardown() {
+	sudo rm -rf $HOST_TMP_TEST_DIR
+	[ -f "$KATA_CONFIG_BACKUP" ] && sudo mv "$KATA_CONFIG_BACKUP" "$KATA_CONFIG" || \
+		sudo cp "$DEFAULT_KATA_CONFIG" "$KATA_CONFIG"
+}
+
+function setup() {
+	HOST_TMP_TEST_DIR="$(mktemp -d)"
+	sudo mkdir -p $(dirname "${KATA_CONFIG}")
+	[ -f "$KATA_CONFIG" ] && sudo cp "$KATA_CONFIG" "$KATA_CONFIG_BACKUP" || \
+		sudo cp "$DEFAULT_KATA_CONFIG" "$KATA_CONFIG"
+}
+
+
+@test "Ensure poststart OCI hooks work" {
+	local tmp_guest_img_path="$HOST_TMP_TEST_DIR/kata-containers.img"
+	local tmp_hooks_filepath="$HOST_TMP_TEST_DIR/hooks.sh"
+	local tmp_mount_point="$HOST_TMP_TEST_DIR/mnt"
+
+	# leveraging an oci poststart hook so that we can inspect the file after the container is up and running.
+	local tmp_oci_hooks_dir="$tmp_mount_point/usr/share/oci/hooks/poststart"
```

Comment:
This is the only line different between this test function and the prestart test function below. I'd consider creating a generic test function and passing "prestart" or "poststart" to it from each of the `@test` functions to avoid all the duplication.